pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -1676,6 +1691,14 @@ func isImmutable(cr *api.KieApp) bool {
 	return false
 }
 
+func isTrial(cr *api.KieApp) bool {
```

Comment:
IsTrial function uses status of `KieApp` CR. CR status will not be available at this point.

Reply:
not really, it is part of the reconciliation loop. 
the CR is available, as well the CR.Applied confs.